Source/Core/Core/PowerPC/PPCTables.cpp

```diff
@@ -55,169 +36,64 @@ const std::array<u64, 16> m_crTable = {{
 
 namespace PPCTables
 {
-GekkoOPInfo* GetOpInfo(UGeckoInstruction inst)
+struct OpDispatch
 {
-  const GekkoOPInfo* info = m_infoTable[inst.OPCD];
-  if (info->type == OpType::Subtable)
-  {
-    switch (inst.OPCD)
-    {
-    case 4:
-      return m_infoTable4[inst.SUBOP10];
-    case 19:
-      return m_infoTable19[inst.SUBOP10];
-    case 31:
-      return m_infoTable31[inst.SUBOP10];
-    case 59:
-      return m_infoTable59[inst.SUBOP5];
-    case 63:
-      return m_infoTable63[inst.SUBOP10];
-    default:
-      ASSERT_MSG(POWERPC, 0, "GetOpInfo - invalid subtable op %08x @ %08x", inst.hex, PC);
-      return nullptr;
-    }
-  }
-  else
-  {
-    if (info->type == OpType::Invalid)
-    {
-      ASSERT_MSG(POWERPC, 0, "GetOpInfo - invalid op %08x @ %08x", inst.hex, PC);
-      return nullptr;
-    }
-    return m_infoTable[inst.OPCD];
-  }
-}
+  std::bitset<64> leaf_flags;
+  std::bitset<64> subtables;
+  u32 op_index_start;
+  u16 dispatch_start;
+  u8 subop_shift;
+  u8 subop_len;
+};
+static std::array<OpDispatch, 30> dispatch_table = {{
+#include "OpID_DecodingTable.gen.cpp"
+}};
+
+static constexpr GekkoOPInfo UNKNOWN = {"Invalid Opcode", OpType::Invalid, 0};
```

Comment:
This can just be `constexpr`, constexpr internally links by default with variables.